cpp/src/sqls/sqls_rtti_comp.h

```diff
@@ -25,45 +22,181 @@
 
 #include "rmm/thrust_rmm_allocator.h"
 
-//for int<n>_t:
-//
 #include <cstdint>
 
-///#include "gdf/cffi/types.h"
+#include "utilities/cudf_utils.h"
+#include "utilities/type_dispatcher.hpp"
 
 template<typename IndexT>
 struct LesserRTTI
 {
-  LesserRTTI(void* const* cols,
-	     int* const types,
-	     size_t sz):
-    columns_(cols),
-    rtti_(types),
-    sz_(sz),
-    vals_(nullptr)
+  LesserRTTI(void *const *cols,

```

Comment:
Instead of having all of these different constructors, could you instead add default values for the various class members? 

e.g.,

`const void* const * vals_{nullptr};`

That way the compiler will take care of filling in the default values for you instead of having to specify all the different constructors where a value may or may not be given for each member variable of the struct.